src/Bravo/Signer.cpp

```diff
@@ -0,0 +1,44 @@
+#include "Signer.h"
+
+#include <TrezorCrypto/ecdsa.h>
+#include <TrezorCrypto/secp256k1.h>
+
+using namespace TW::Bravo;
+
+void Signer::sign(const PrivateKey& privateKey, Transaction& transaction, int(*canonicalChecker)(uint8_t by, uint8_t sig[64])) const {
+    if (!transaction.isValid()) {
+        throw std::invalid_argument("Invalid Transaction!");;
+    }
+
+    Data result;
+    result.resize(65);
+    if (ecdsa_sign_digest(&secp256k1, privateKey.bytes.data(), hash(transaction).data(), result.data() + 1, result.data(), canonicalChecker) != 0) {
```

Comment:
try `privateKey.sign`

Reply:
For Bravo, there are two change in passing parameter of call `ecdsa_sign_digest`.
- Canonical check is required which is `NULL` by default in `privateKey.sign`. I didn't want to change that implementation.
-  Bravo stores recovery Id on the first byte instead of last.


Any suggestion?

Reply:
you can add another `privateKey.sign` method which requires canonical parameter

Reply:
okay